google/cloud/security/common/gcp_api/compute.py

```diff
@@ -72,7 +140,28 @@ def get_backend_services(self, project_id):
                     previous_request=list_request,
                     previous_response=response)
         except (HttpError, HttpLib2Error) as e:
-            raise api_errors.ApiExecutionError('backend_services', e)
+            raise api_errors.ApiExecutionError(resource_name, e)
+
+    def get_backend_services(self, project_id):
+        """Get the backend services for a project.
+
+        Args:
+            project_id: The project id.
+
+        Yield:
```

Comment:
@blueandgold and I discussed this offline.  By using the _flatten_list / _flatten_aggregated list, if there's an APIError exception raised by retrieving *any* page, the pipeline won't load *any* of the items.  With the previous approach where we returned an iterator and did the flattening inside the pipeline code, only the single page that raised an error would be omitted but the pipeline loader would log and continue fetching the next page.

That is, if there is an error when retrieving a page, then the pipeline won't load any data for a project, but the pipeline will move to the next project. 

If we care about getting partial results, the APIError exception can be handled in the _flatten_list / _flatten_aggregated list methods instead. However, we decided to leave it as-is for now because:

* Converting from generators to simple lists earlier rather than later simplifies the code.
* Most errors, like access denied, will happen on the first page (given that base_client already retries on transient errors like connection refused.)

If we get user reports of frequent non-first-page errors we can revisit.